staging/src/k8s.io/client-go/discovery/discovery_client.go

```diff
@@ -456,12 +459,14 @@ func setDiscoveryDefaults(config *restclient.Config) error {
 	if config.Timeout == 0 {
 		config.Timeout = defaultTimeout
 	}
-	if config.Burst == 0 && config.QPS < 100 {
+	// if a burst limit is not already configured and
+	// an avg. rate of `defaultBurst` qps is not already configured
+	if config.Burst == 0 && config.QPS < defaultBurst {
```

Comment:
The condition here is a bit weird, since if burst is already set to something like 2, it won't get fixed?

And if burst is set to zero, likely qps will also be zero, and then we don't modify qps?

Reply:
hrmm... trying to page that back in...

Reply:
I think the intent was two-fold:
1. this should be a default, so if Burst is explicitly set we don't want to override it
2. I vaguely remember the burst bucket defaulting to match QPS at some layer (at least at some point in time), so the `config.QPS < 100` guard was to prevent us from defaulting to a burst value *lower* than what would have been auto-selected. I'm not sure if this guard is still necessary

Reply:
Thanks for the clarification. Kept the `config.Burst == 0` part of the conjunction and removed the `config.QPS < defaultBurst` condition.